lib/ace/puppet_util.rb

```diff
@@ -23,11 +25,12 @@ def self.init_global_settings(ca_cert_path, ca_crls_path, private_key_path, clie
       Puppet.settings[:plugindest] = File.join(cachedir, 'plugins')
       Puppet.push_context(Puppet.base_context(Puppet.settings), "Puppet Initialization")
       # ssl_context will be a persistent context
+      cert_provider = Puppet::X509::CertProvider.new
```

Comment:
push the paths into the constructor (https://github.com/puppetlabs/puppet/blob/master/lib/puppet/x509/cert_provider.rb#L14) and use `load_cacerts(required: true)` below instead. no need to do the IO ourselves either.

Reply:
sad that the client_cert and private_key methods require a `name` instead of a filename, but we won't be able to change that in 6.4.1, so we'll have to make do for now